docs/upgrading_wagtail_v2/index.rst

```diff
@@ -0,0 +1,158 @@
+       $ wagtail updatemodulepaths /path/to/project
+
+For the full list of command line options, enter ``wagtail help updatemodulepaths``.
+
+You are advised to take a backup of your project codebase before running this command. The command will perform a search-and-replace over all \*.py files for the affected module paths; while this should catch the vast majority of module references, it will not be able to fix instances that do not use the dotted path directly, such as ``from wagtail import wagtailcore``.
+
+Upgrading directly to Wagtail 2:
+================================
+**WARNING:** You should first locally try to upgrade version by version to see if there are any major application issues before attempting to upgrade directly. 
+
+----------------
+Tips and Tricks:
+----------------
+During the deployment process, you may find that you want to try and upgrade directly to Wagtail 2 without having to do multiple deployments. In addition, when running python tests for migrations, your migrations will most likely fail as your test database is a new blank database that tests run from. They fail because your home migrations run a dependency that reference a Wagtail app migration that has not yet run. 
+
+Locally upgrade environment:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+1. Upgrade version of Wagtail to v2 and Django in the requirements.txt
+2. Run the update script: ``wagtail updatemodulepaths``
+3. Run migrations ``./manage.py migrate``. This will go ahead and add all of the necessary migrations to your database. You won't need to run these migrations again.
+4. Run tests ``./manage.py test``. You may find that migrations may fail at a specific migration, take note of that.
+5. Add the failed migration from tests to the initial home migration dependency. To track down the correct app and run ``./manage.py showmigrations``. 
+6. Run tests again ``./manage.py test``. At this point, tests should pass.
+
+Deployment:
+~~~~~~~~~~~
+1. Create one branch that upgrades and disables migrations from python tests. Merge this first to deploy upgrade.
+2. Create another branch that adds the necessary failed test dependency to your intial home migration. Merge this second and deploy. 
+
+After following the 2 deployment steps above, that will upgrade you to the most recent version of Wagtail with passing tests and migrations. You wont have to do this again unless there is another major upgrade to wagtail.
+
+We've Upgraded...what now?:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* Take a good look at all of your custom built applications to see if they still work in Wagtail 2.
+* Hallo vs Draftail: Since you cannot run both rich text editor libraries at the same time, you will need to choose between using Hallo or Draftail. HalloPlugins have been deprecated, and will be removed in Wagtail 1.14. So it would be best that you make the effort to attempt to transfer over to draftail now, this comes by default with Wagtail 2. 
+
+Compatible Django / Python versions
```

Comment:
Can this link to `docs/releases/upgrading.rst#compatible-django--python-versions` instead so we don't have to maintain the lists of compatible versions in 2 places?